src/__tests__/requirement12.test.js

```diff
@@ -22,10 +13,13 @@ describe(`12 - Finalize a compra vendo um resumo dela, preenchendo os seus dados
     const phone = '99999999999';
     const cep = '99999999';
     const address = 'my address is where I live';
+    jest.spyOn(global, 'fetch').mockImplementation((url) => {
+      return mockFetch(url);
+    });
     render(<App />);
-    await waitFor(() => expect(api.getCategories).toHaveBeenCalled());
+    await waitFor(() => expect(global.fetch).toHaveBeenCalled());
     fireEvent.click(screen.getAllByTestId('category')[0]);
-    await waitFor(() => expect(api.getProductsFromCategoryAndQuery).toHaveBeenCalled());
+    await waitFor(() => expect(global.fetch).toHaveBeenCalled());
```

Comment:
Mesma situação aqui :dog: 

```suggestion
    await waitFor(() => expect(global.fetch).toHaveBeenCalledTimes(2));
```